### PR TITLE
Implement URL tracking parameters protection feature

### PR DESCRIPTION
**Pull request by kzar**
Implement the URL tracking parameters protection feature. The feature
strips known tracking parameters for navigations, assuming that
neither the initiator URL (referrer) nor request URL (destination) are
allowlisted.

See also the reference tests[1] and test pages[2] for more details.

1 - https://github.com/duckduckgo/privacy-reference-tests/tree/main/url-parameters
2 - https://privacy-test-pages.glitch.me/privacy-protections/query-parameters

**Reviewer:** @SlayterDev 
**CC:** @kdzwinel 

## Automated tests:
- [x] Unit tests
- [x] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/url-parameters.es6.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+function ensureTrackingParametersConfig () {
+    if (plainTextTrackingParameters && regexpTrackingParameters) {
+        return true
+    }
+
+    if (!tdsStorage.config ||
+        !tdsStorage.config.features ||
+        !tdsStorage.config.features.trackingParameters ||
+        !tdsStorage.config.features.trackingParameters.settings ||
+        !tdsStorage.config.features.trackingParameters.settings ||
```
Comment (SlayterDev) [inline shared/js/background/url-parameters.es6.js:24]: This line appears to be duplicated?

Comment (SlayterDev) [reply shared/js/background/url-parameters.es6.js:24]: Additionally can this be cleaned up using optional chaining (Or whatever its called in JS)?

e.g. `const trackingParams = tdsStorage.config?.features?.trackingParameters?.settings || null`

**Review by SlayterDev — CHANGES_REQUESTED** [review]
Dave, looking good so far. Will test things out but just have one clean up item for you to look at.

File: shared/js/background/url-parameters.es6.js (hunk 1 above)
Comment (kzar) [reply shared/js/background/url-parameters.es6.js:24]: > This line appears to be duplicated?

Oops, well spotted. Done.

> Additionally can this be cleaned up using optional chaining...

Unfortunately that syntax [isn't supported by Firefox < 74](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility), but we support Firefox 65+.

**Review by SlayterDev — APPROVED** [review]
@kzar works great! In the future it may be easier for reviewers to branch and push directly to this repo rather than using your own fork. That way they don't have to add your fork as an origin 👍

**Pull request merged**